kge/util/dump.py

```diff
@@ -113,90 +113,174 @@ def _add_dump_trace_parser(subparsers_dump):
     parser_dump_trace = subparsers_dump.add_parser(
         "trace",
         help=(
-            "Process and dump trace to stdout and/or csv. The trace will be processed "
-            "backwards, starting with a specified job_id."
+            "Dump trace to csv (default) and/or stdout. The tracefile will be processed"
+            " backwards starting from the last entry. Further options allow to start"
+            " processing from a particular checkpoint or job_id or epoch_number."
         ),
     )
-
     parser_dump_trace.add_argument(
         "source",
         help="A path to either a checkpoint or a job folder.",
         nargs="?",
         default=".",
     )
-
+    parser_dump_trace.add_argument(
+        "--train",
+        action="store_const",
+        const=True,
+        default=False,
+        help=(
+            "Include entries from training jobs (enabled when none of --train, --valid,"
+            " or --test is specified)."
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--valid",
+        action="store_const",
+        const=True,
+        default=False,
+        help=(
+            "Include entries from validation during training and evaluation on"
+            " the validation data split (enabled when none of --train, --valid, or"
+            " --test is specified)."
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--test",
+        action="store_const",
+        const=True,
+        default=False,
+        help=(
+            "Include entries from evaluation on the test data split (enabled when "
+            "  none of --train, --valid, or --test is specified)."
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--search",
+        action="store_const",
+        const=True,
+        default=False,
+        help="Dump the tracefile of a search job. The options --train, --valid, and"
+             " --test are not applicable.",
+    )
+    parser_dump_trace.add_argument(
+        "--keysfile",
+        default=False,
+        help=(
+            "A path to a file which contains lines in the format"
+            " 'new_key_name'='key_name'. For every line in the keys file, the command"
+            " will look for the value of 'key_name' in the trace entries and config and"
```

Comment:
trace entries (first) and config (second)

Reply:
I think you mean it the other way around?

Reply:
Trace is tried first, right (it should be)

Reply:
For the keys from --keysfile and --keys, at the moment, the config is searched and then the trace entry is searched. So then this has to be changed?

Reply:
Yes, that's unintenional; please change. It probably only affects a small number of keys though.

Reply:
Okey I will change that.